src/main/java/org/zeplinko/commons/lang/ext/core/AbstractResult.java

```diff
@@ -0,0 +1,129 @@
+
+    /**
+     * Returns the data if the AbstractResult is successful; otherwise, returns the
+     * other specified value.
+     *
+     * @param other The value to return if the AbstractResult is a failure.
+     * @return The success data, or {@code other} if the AbstractResult is a
+     *         failure.
+     */
+    public D orElse(D other) {
+        return this.isFailure() ? other : this.getData();
+    }
+
+    /**
+     * Returns the data if the AbstractResult is successful; otherwise, applies the
+     * provided supplier to get the other value.
+     *
+     * @param otherSupplier The supplier function to provide the other value if the
+     *                      AbstractResult is a failure.
+     * @return The success data, or the value provided by {@code otherSupplier} if
+     *         the AbstractResult is a failure.
+     */
+    public D orElseGet(@Nonnull Supplier<? extends D> otherSupplier) {
+        Objects.requireNonNull(otherSupplier);
+        return this.isFailure() ? otherSupplier.get() : this.getData();
+    }
+
+    /**
+     * Returns the data if the AbstractResult is successful; otherwise, applies the
+     * provided function to the error and returns the AbstractResult.
+     *
+     * @param otherFunction The function to apply to the error if the AbstractResult
+     *                      is a failure.
+     * @return The success data, or the AbstractResult of applying
+     *         {@code otherFunction} to the error.
+     */
+    public D orElseGet(@Nonnull Function<? super E, ? extends D> otherFunction) {
+        Objects.requireNonNull(otherFunction);
+        return this.isFailure() ? otherFunction.apply(this.getError()) : this.getData();
+    }
+
+    /**
+     * Returns the data if the AbstractResult is successful; otherwise, throws the
+     * exception provided by the function applied to the error.
+     *
+     * @param exceptionFunction The function to apply to the error to generate an
+     *                          exception.
+     * @param <X>               The type of the exception to be thrown if the
+     *                          AbstractResult is a failure.
+     * @return The success data.
+     * @throws X if the AbstractResult is a failure, the exception generated by
+     *           applying {@code exceptionFunction} to the error.
+     */
+    public <X extends Throwable> D orElseThrow(@Nonnull Function<? super E, ? extends X> exceptionFunction) throws X {
+        Objects.requireNonNull(exceptionFunction);
+        if (this.isFailure()) {
+            throw exceptionFunction.apply(this.getError());
+        }
+        return this.getData();
+    }
+
+    /**
+     * Returns the Optional with value if the AbstractResult is successful;
+     * otherwise, returns the Empty Optional.
+     *
+     * @return Optional with value if the AbstractResult is successful, otherwise
+     *         Empty Optional
+     */
+    public Optional<D> toOptional() {
```

Comment:
😄 This gives me the `toJava()` vibes from `vavr`
https://github.com/vavr-io/vavr/blob/36f6655fc03870963985e77d89a3b82d4366fb6b/src/main/java/io/vavr/collection/LinkedHashMap.java#L958

Reply:
Yeah